### ISSUE TITLE
Unable to create okta_group from okta_user datasource input

### ISSUE THREAD

**Issue by 007** [okta/terraform-provider-okta#310]
tl;dr: Please surface Okta's internal `id` as an attribute of `okta_user` lookups (call it `index` to match the `okta_user` resource), or allow `okta_group` to be populated via email.

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.13.5
+ provider registry.terraform.io/oktadeveloper/okta v3.7.4
```

### Affected Resource(s)

- okta_user
- okta_group

### Terraform Configuration Files

```hcl
data "okta_users" "me" {
  search {
    name = "profile.login"
    comparison = "eq"
    value = "my.email@example.com"
  }
}

resource "okta_group" "test" {
  name = "test-group"
  users = data.okta_users.me
}

```
There is no attribute or permutation of `okta_users` that works to populate this list. It doesn't matter if I substitute `okta_user` in place of `okta_users` or if I rewrite the return values via `for` loop.

### Expected Behavior

I should be able to use email to populate an `okta_group`, or I should be able to obtain an `id` from an `email` via some search criteria.

### Actual Behavior

Unable to create a group based on anything other than `id` and unable to obtain `id` via lookup. `okta_users` does present an `id` for the users that it returns, but that `id` seems to be arbitrary and not related to the Okta `id`.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

The lookup API that is used for `okta_user` matching is returning the fields that would be required:
```
$COMPANY.oktapreview.com/api/v1/users?limit=200&sortOrder=0&search=profile.login+eq+%22$EMAIL_GOES_HERE%22
```

```
[
 {
  "id": "REDACTED_THIS_IS_THE_ID_THAT_I_NEED",
  "status": "ACTIVE",
  "created": "2021-01-29T00:49:49.000Z",
  "activated": "2021-01-29T00:49:49.000Z",
  "statusChanged": "2021-01-29T18:39:07.000Z",
  "lastLogin": "2021-01-29T23:20:27.000Z",
  "lastUpdated": "2021-01-29T18:39:07.000Z",
  "passwordChanged": "2021-01-29T18:39:07.000Z",
  "type": {
   "id": "REDACTED"
  },
  "profile": {
   "firstName": "Ryan",
   "lastName": "Moore",
   "mobilePhone": null,
   "secondEmail": "",
   "login": "redacted@redacted",
   "email": "redacted@redacted"
  },
...
```

The output from the terraform resource includes this `id` but does not include the Okta ID:
```
{
  "redacted@redacted" = {
    "id" = "3566712942"
    "search" = [
      {
        "comparison" = "eq"
        "name" = "profile.login"
        "value" = "redacted@redacted"
      },
    ]
    "users" = [
      {
        "admin_roles" = []
        "city" = ""
        "cost_center" = ""
        "country_code" = ""
        "custom_profile_attributes" = "{}"
        "department" = ""
        "display_name" = ""
        "division" = ""
        "email" = "redacted@redacted"
        "employee_number" = ""
        "first_name" = "Ryan"
        "group_memberships" = []
        "honorific_prefix" = ""
        "honorific_suffix" = ""
        "last_name" = "Moore"
        "locale" = ""
        "login" = "redacted@redacted"
        "manager" = ""
        "manager_id" = ""
        "middle_name" = ""
        "mobile_phone" = ""
        "nick_name" = ""
        "organization" = ""
        "postal_address" = ""
        "preferred_language" = ""
        "primary_phone" = ""
        "profile_url" = ""
        "second_email" = ""
        "state" = ""
        "status" = "ACTIVE"
        "street_address" = ""
        "timezone" = ""
        "title" = ""
        "user_type" = ""
        "zip_code" = ""
      },
    ]
  }
}
```

**Comment by bogdanprodan-okta** [comment]
Hi, @007! Thanks for creating this issue! I've already added the `id` attribute into the `okta_users` data source and it will be available in the upcoming v3.8.0. Check the code [here](https://github.com/oktadeveloper/terraform-provider-okta/blob/master/okta/data_source_okta_users.go#L48).  
P.S. New version will probably be released on Monday.
